### PR TITLE
Problem: (CRO-577) Client allows weak passphrases

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Estimate password strength using `zxcvbn` crate. Return `Error` when password score is less than 3.

### CODE REVIEW

File: client-core/src/wallet/default_wallet_client.rs [hunk 1]
```diff
@@ -620,3 +636,31 @@ where
...
+            }
+
+            for suggestion in feedback.suggestions() {
+                feedback_str.push_str(&format!("Suggestion: {} | ", suggestion));
+            }
+
+            if feedback_str.is_empty() {
+                feedback_str = "No feedback available!".to_string();
+            } else {
+                // Pop last three chars
```
Comment (yihuang) [inline client-core/src/wallet/default_wallet_client.rs]: I guess we can (join)[https://doc.rust-lang.org/1.30.0/std/slice/trait.SliceConcatExt.html#tymethod.join] here.

Comment (devashishdxt) [reply client-core/src/wallet/default_wallet_client.rs]: Done

**Comment by devashishdxt** [comment]
bors try

File: Cargo.lock [hunk 2]
```diff
@@ -184,6 +184,19 @@ dependencies = [
...
+source = "registry+https://github.com/rust-lang/crates.io-index"
+dependencies = [
+ "bit-vec 0.5.1 (registry+https://github.com/rust-lang/crates.io-index)",
+]
+
+[[package]]
+name = "bit-vec"
+version = "0.5.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+
```
Comment (tomtau) [inline Cargo.lock]: this one is a bit annoying / bloaty, as there's already `bit-vec 0.6.1`  (and `bitvec`).

probably open an issue for another dependency review / cleanup

File: client-core/src/wallet/default_wallet_client.rs [hunk 3]
```diff
@@ -102,6 +103,21 @@ where
...
+        let password_entropy = estimate_password_strength(passphrase.unsecure(), &[name])
+            .chain(|| (ErrorKind::IllegalInput, "Blank passphrase"))?;
+
+        if password_entropy.score() < 3 {
+            return Err(Error::new(
+                ErrorKind::IllegalInput,
+                format!(
+                    "Weak passphrase: {}",
+                    parse_feedback(password_entropy.feedback().as_ref())
+                ),
```
Comment (tomtau) [inline client-core/src/wallet/default_wallet_client.rs]: I guess this is probably fine, but not sure how annoying it'd be in quick testing -- we'll see, perhaps later there could be a "unsafe_development_only" flag to bypass that

**Comment by devashishdxt** [comment]
bors try

File: client-core/src/wallet/default_wallet_client.rs (hunk 3 above)
Comment (devashishdxt) [reply client-core/src/wallet/default_wallet_client.rs]: We can add following code in future to only build this in release builds:
```rust
if !cfg!(debug_assertions) {
    // Check password score only in non-debug builds
}
```

**Comment by devashishdxt** [comment]
bors try

**Comment by devashishdxt** [comment]
bors try

**Pull request merged**